### PR TITLE
[TESTING] CudaNext CMake + CI

### PR DESCRIPTION
**Pull request by alliepiper**
Just testing CI / CMake changes without disturbing the original PR (#1485)

### CODE REVIEW

File: .github/actions/compute-matrix/compute-matrix.sh [hunk 1]
```diff
@@ -27,6 +32,11 @@ extract_matrix() {
   write_output "PER_CUDA_COMPILER_MATRIX"  "$per_cuda_compiler_matrix"
   write_output "PER_CUDA_COMPILER_KEYS" "$(echo "$per_cuda_compiler_matrix" | jq -r 'keys | @json')"
 
+  local nvcc_full_matrix_minstd_17="$(echo "$matrix" | jq -cr '.nvcc' | filter_minstd_17 | explode_std_versions)"
```
Comment (jrhemstad) [inline .github/actions/compute-matrix/compute-matrix.sh:35]: suggestion: Alternatively, we could just define a separate sub-matrix for CUDA next testing. That would let us limit which host compilers we test it with as well.

Comment (alliepiper) [reply .github/actions/compute-matrix/compute-matrix.sh:35]: Let's discuss at the meeting. I'm going off of the assumption that we want to treat it like our other libraries with the same level of testing, but if we need to reduce coverage that's definitely an option.

**Comment by alliepiper** [comment]
Some discussion from our meeting:

- Users should have to opt in.
  - Install rules off by default
  - add_subdirectory/find_package on CCCL shouldn't add CudaNext targets without explicit opt-in
- Some planned features may only be on newer CTK. For now, minimum CTK version is 12.3. CI should be updated.
- Leave compiler matrix as is for now. We can reduce coverage if needed as the dev team digs into the failing builds.
- Header locations + namespace + proj name? TBD
- File extensions? .cuh for CUDA, .h for host-only

**Pull request closed without merging**